api/v1alpha1/provider_types.go

```diff
@@ -20,7 +20,7 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
-	ctrlconfigv1 "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+	ctrlconfigv1 "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1" //nolint:staticcheck
```

Comment:
not big fan of adding nolints to the api folder, maybe there is a way we can overcome it by adding `exclude-rules` to `staticcheck` linter in golangci.yml, as I see multiple examples in https://github.com/kubernetes-sigs/cluster-api/blob/main/.golangci.yml#L221-L236?